tests/multimodel/test_adex.py

```diff
@@ -0,0 +1,316 @@
+        adex = self._create_exc_mass()
+        adex.update_params(UPDATE_PARAMS)
+        self.assertEqual(adex.params["tau_m"], 15.0)
+        self.assertEqual(adex.params["c_global"], 0.4 * adex.params["tau_syn_exc"] / 3.0)
+
+    def test_run(self):
+        adex_exc = self._create_exc_mass()
+        adex_inh = self._create_inh_mass()
+        for adex in [adex_exc, adex_inh]:
+            result = self._run_node(adex, DURATION, DT)
+            self.assertTrue(isinstance(result, np.ndarray))
+            self.assertTupleEqual(result.shape, (int(DURATION / DT), adex.num_state_variables))
+
+
+class TestAdExNode(unittest.TestCase):
+    def _create_node(self):
+        node = AdExNode(exc_seed=SEED, inh_seed=SEED)
+        node.index = 0
+        node.idx_state_var = 0
+        node.init_node()
+        return node
+
+    def test_init(self):
+        adex = self._create_node()
+        self.assertTrue(isinstance(adex, AdExNode))
+        self.assertEqual(len(adex), 2)
+        self.assertDictEqual(_strip_keys(adex[0].params), _strip_keys(DEFAULT_PARAMS_EXC))
+        self.assertDictEqual(_strip_keys(adex[1].params), _strip_keys(DEFAULT_PARAMS_INH))
+        self.assertTrue(hasattr(adex, "_rescale_connectivity"))
+        self.assertEqual(len(adex._sync()), 4 * len(adex))
+        self.assertEqual(len(adex.default_network_coupling), 2)
+        np.testing.assert_equal(
+            np.array(sum([adexm.initial_state for adexm in adex], [])), adex.initial_state,
+        )
+
+    def test_update_rescale_params(self):
+        adex = self._create_node()
+        # update connectivity and check rescaling
+        old_rescaled = adex.connectivity.copy()
+        adex.update_params({"local_connectivity": 2 * DEFAULT_ADEX_NODE_CONNECTIVITY})
+        np.testing.assert_equal(adex.connectivity, 2 * old_rescaled)
+
+    def test_run(self):
+        adex = self._create_node()
+        all_results = []
+        for backend, noise_func in BACKENDS_TO_TEST.items():
+            result = adex.run(
+                DURATION, DT, noise_func(ZeroInput(DURATION, DT, adex.num_noise_variables)), backend=backend
+            )
+            self.assertTrue(isinstance(result, xr.Dataset))
+            self.assertEqual(len(result), adex.num_state_variables)
+            self.assertTrue(all(state_var in result for state_var in adex.state_variable_names[0]))
+            self.assertTrue(
+                all(result[state_var].shape == (int(DURATION / DT), 1) for state_var in adex.state_variable_names[0])
+            )
+            all_results.append(result)
+        # test results are the same from different backends
+        for state_var in all_results[0]:
+            corr_mat = np.corrcoef(
+                np.vstack([result[state_var].values.flatten().astype(float) for result in all_results])
+            )
+            self.assertTrue(np.greater(corr_mat, CORR_THRESHOLD).all())
+
+    def test_compare_w_neurolib_native_model(self):
```

Comment:
@caglarcakan here is the comparison between native ALN and this one for one node